### ISSUE TITLE
SocialSharing doesn't work iOS 9

### ISSUE THREAD

**Issue by moisesbsfilho** [EddyVerbruggen/SocialSharing-PhoneGap-Plugin#669]
I'm using cordova 6.3.1 version and I have a problem in iOS application. When I run my application and I press the share button nothing happens. But if I dispose to second plan and return to first plan it's work with a console message: Plugin should use a background thread! I don't know what to do. I think that plugin is in a blocking state.

**Comment by logikamente** [comment]
I fixed it just now!
I tried to upgrade plugin latest.

`cordova plugin add cordova-plugin-x-socialsharing`
`cordova prepare`

It work for my ios 9 app

**Comment by EddyVerbruggen** [comment]
@moisesbsfilho first plan / second plan?

@logikamente are you responding to this thread?

**Comment by moisesbsfilho** [comment]
When I start my app and press share button nothing happens, there is nothing in log and doesn't appear the Facebook App share's view. But if I put app in background (open another app or just minimize the application) and I turn it to foreground in view, the Facebook's share appear to me. But in console show: Plugin should use a background thread!

1 - Pressing share button
![img-20160829-wa0005](https://cloud.githubusercontent.com/assets/6910311/18059841/4bf72db6-6df1-11e6-8f9e-0941c69b82df.jpg)

2 - Turning it to background
![img-20160829-wa0003](https://cloud.githubusercontent.com/assets/6910311/18059850/54cfac06-6df1-11e6-9a46-b3f601969aac.jpg)

3 - Returning it to foreground
![img-20160829-wa0004](https://cloud.githubusercontent.com/assets/6910311/18059865/70f7d584-6df1-11e6-802d-a32117c7148e.jpg)

**Comment by EddyVerbruggen** [comment]
Very strange, I'd like to make sure it's not smething in your app that's the problem, so can you please `cordova create testapp`, add iOS, add this plugin, and add a share button similar to what your app has?

**Comment by moisesbsfilho** [comment]
Eddy, thank's! My problem it was my config.xml. It was wrong! I included this lines:

`<plugin name="cordova-plugin-whitelist" spec="1" />
 <access origin="*" />
<allow-intent href="http://*/*" />
<allow-intent href="https://*/*" />
<allow-intent href="tel:*" />
<allow-intent href="sms:*" />
<allow-intent href="mailto:*" />
<allow-intent href="geo:*" />

<platform name="ios">
     <allow-intent href="itms:*" />
     <allow-intent href="itms-apps:*" />
</platform>
`

**Comment by ebhoren** [comment]
I'm having the same bug on iOS 10 using Phonegap 6.3.4.
When pushing my share button, it trigger **window.plugins.socialsharing.shareViaFacebook('Test', null, 'http://www.google.com');** Nothing happens.

When i put my app in background (open another app or just minimize the application) and I turn it to foreground in view, the Facebook's share appear to me. 

I've tried @moisesbsfilho solution but it didn't work.
I've also tried to create a fresh new project and the same problem occur.

```
phonegap create testapp
cd testapp/
phonegap plugin add cordova-plugin-x-socialsharing
phonegap platform add ios
phonegap prepare
```

Am i doing something wrong?
Thanks
